### DISCUSSION TITLE
Microsoft releases .NET 5, and the future of CSLA

### DISCUSSION THREAD

**Discussion by rockfordlhotka** [MarimerLLC/csla#1896] [General]
Very exciting day - Microsoft has released .NET 5!

I'm busily working on getting CSLA all updated to the released versions of Visual Studio, .NET 5, and also moving any remaining .NET Core 3.0 code to target 3.1 (because 3.0 just became officially obsolete).

In terms of guidance, remember that .NET 5 is _not_ an LTS (long term support) release of .NET. That means it, like .NET Core 3.0, will immediately become unsupported on the day that .NET 6 comes out (in about a year).

So if you want to avoid version churn, and are OK not using the latest and greatest features, you should stick with .NET Core 3.1.

If you want the latest and greatest features, you can target .NET 5, but be aware that you'll _need to update_ to .NET 6 as soon as it is available.

My intent with CSLA is to support LTS and current versions of .NET. Of course _right now_ CSLA still supports all the way back to .NET 4.0!!

But CSLA 6 (next year) will drop support for all super-old technologies, and the plan is to support only:

* .NET Framework 4.8 (mostly via .NET Standard 2.0 like today)
* .NET 6

Any other dependent frameworks, like old versions of ASP.NET, EF, etc. will be dropped if they can't run on one of those two versions of .NET.

**Comment by rockfordlhotka** [comment]
Also, the appveyor build is currently broken, because appveyor hasn't yet got their container images updated for the just-released Visual Studio and .NET 5. I expect the build to "just start working" within the next couple days.

In the meantime, I'm continuing to work and will trust that if the build/tests run on my workstation that I can move forward 😄

**Reply by BlagoCuljak** [reply]
Hi Rocky,
When we were migrating from .net to .net core, we had great help from this tool, and of course, you can check if upgrade to NET5 is easier said then done. 

[https://docs.microsoft.com/en-us/dotnet/standard/analyzers/portability-analyzer](https://docs.microsoft.com/en-us/dotnet/standard/analyzers/portability-analyzer)

**Reply by rockfordlhotka** [reply]
Thank you Blago - the issue with appveyor is simply that they need to update their base container images for their CI agent so it has the final release of VS and .NET 5.
